components/credentials/src/main/kotlin/org/cloudfoundry/credhub/entity/Credential.kt

```diff
@@ -21,6 +24,13 @@ class Credential : AuditableCredential {
     @GenericGenerator(name = "uuid2", strategy = "uuid2")
     override var uuid: UUID? = null
 
+    @OneToMany(
+        cascade = [CascadeType.REMOVE],
+        mappedBy = "credential", fetch = FetchType.EAGER
+    )
+    var credentialVersions: MutableList<CredentialVersionData<*>> =
+        mutableListOf()
```

Comment:
Does this mean that a new column will be added to the `credential` table, storing all of the credential versions associated to it?

Reply:
No. That relationship is already in the database table schema via the FK from `credential_version` to `credential`. This JPA layer causes no change to the existing database schema. Otherwise there will be an issue for the credhub upgrade. Verified it with both fresh credhub database and upgraded credhub database.

Reply:
Oh that's great. So what's the impact of this change? Will the server need to keep more data in memory? Like a `Map<Credential, List<Version>>` of some sort? Or is this again all dealt with through the DB's FKs?

It looks really magic. You just added this field, it appears to be never used anywhere, but it fixes the issue we've been having?

Reply:
> Will the server need to keep more data in memory? Like a Map<Credential, List<Version>> of some sort?

Likely so to some degree.

> It looks really magic. 

There is no magic. Just more understanding.

> but it fixes the issue we've been having?

Right, that is the whole point of this PR.

Reply:
Actually, there is a small magic that surprised me. I thought that there would be issues deleting `encrypted_value` associated with `user_credential` or `password_credential`, but there were all working fine, too.